compose/service.py

```diff
@@ -246,6 +246,12 @@ def recreate_container(self, container, **override_options):
         self.client.rename(
             container.id,
             '%s_%s' % (container.short_id, container.name))
+
+        override_options = dict(
+            override_options,
+            environment=merge_environment(
+                override_options.get('environment'),
+                {'affinity:container': container.id}))
```

Comment:
Oh, I did add a test for it, but I didn't realize it required the `==`.

If I set the value to `=<id>` that should work, right ?